lib/Bio/EnsEMBL/DataCheck/Checks/CheckTableSizes.pm

```diff
@@ -0,0 +1,84 @@
+=head1 LICENSE
+
+Copyright [2018-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::CheckTableSizes;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::Registry;
+use Bio::EnsEMBL::Utils::ConfigRegistry;
+use Bio::EnsEMBL::DBSQL::DBAdaptor;
+use Bio::EnsEMBL::Compara::DBSQL::DBAdaptor;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+use Bio::EnsEMBL::DataCheck::Utils qw/ array_diff /;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'CheckTableSizes',
+  DESCRIPTION    => 'Tables must be populated and not differ significantly in row numbers',
+  GROUPS         => ['compara', 'compara_tables'],
+  DATACHECK_TYPE => 'advisory',
+  DB_TYPES       => ['compara'],
+  TABLES         => []
+};
+
+sub tests {
+  my ($self) = @_;
+  
+  my $reg_conf = $self->registry_file;
+  if (defined $reg_conf) {
+    Bio::EnsEMBL::Registry->load_all($reg_conf);
+  }
+  else {
+      die "A compara production_reg_conf needs to be supplied";
+  }
+  
+  my $curr_dba = $self->dba;
+  my $curr_helper = $curr_dba->dbc->sql_helper;
+  my $prev_dba = Bio::EnsEMBL::Compara::DBSQL::DBAdaptor->go_figure_compara_dba('compara_prev');
+  my $prev_helper = $prev_dba->dbc->sql_helper;
+  
+  my $table_sql = "SHOW TABLES";
+  my $curr_tables = $curr_helper->execute_simple( -SQL => $table_sql );
+  my $prev_tables = $prev_helper->execute_simple( -SQL => $table_sql );
+  my $curr_db_name = $curr_dba->dbc->dbname;
+  my $prev_db_name = $prev_dba->dbc->dbname;
+  
+  foreach my $table ( @$curr_tables ) {
+      my $desc_1 = "The number of rows in $table for $curr_db_name has not increased by >10% from $prev_db_name";
+      my $desc_2 = "The number of rows in $table for $curr_db_name has not decreased by >5% from $prev_db_name";
+      my $sql = qq/
+        SELECT COUNT(*) FROM $table
+      /;
+      row_totals($prev_dba, $curr_dba, $sql, $sql, 0.9, $desc_1);
+      row_totals($curr_dba, $prev_dba, $sql, $sql, 0.95, $desc_2);
+  }
+  
+  my $desc = "There are no tables missing in $curr_db_name that were present in $prev_db_name";
+  cmp_ok( scalar(@$curr_tables), '=>', scalar(@$prev_tables), $desc );
+  if ( scalar(@$curr_tables) != scalar(@$prev_tables) ) {
+    diag explain array_diff( $curr_tables, $prev_tables, $curr_db_name, $prev_db_name );
```

Comment:
> `    array_diff
        array_diff($array_1, $array_2, [$array_1_label, [$array_2_label]]);
 
        Calculate the difference between two arrays of string values,
        $array_1 and $array_1, ignoring the order of the elements. This
        method does not work with elements that are references. The
        $array_1_label and $array_2_label parameters allow elements that are
        only in one array to be usefully labelled in the results.`

I was hoping that is what the documentation means. Otherwise, `SHOW TABLES;` does list in alphabetical order by default